rest/src/main/kotlin/builder/interaction/InputChatBuilders.kt

```diff
@@ -114,31 +158,45 @@ public class ChatInputCreateBuilder(
             type,
             Optional.Value(description),
             state.options.mapList { it.toRequest() },
+            state.defaultMemberPermissions,
+            state.dmPermission,
             state.defaultPermission
         )
 
     }
 
 }
 
+@KordDsl
+public interface ChatInputModifyBuilder : ApplicationCommandModifyBuilder, RootInputChatBuilder {
+    public var description: String?
+}
 
 @KordDsl
-public class ChatInputModifyBuilder : ApplicationCommandModifyBuilder, RootInputChatBuilder {
+public interface GlobalChatInputModifyBuilder : ChatInputModifyBuilder, GlobalApplicationCommandModifyBuilder
+
+@KordUnsafe
+public class ChatInputModifyBuilderImpl : GlobalChatInputModifyBuilder {
```

Comment:
Can we make a factory function for this instead? `object : Parent`

Reply:
How would that help?

Reply:
we can make this internal when we don't create this builder in core but call the functions of the service that take builder-lambdas instead

Reply:
We can also make the class internal if we don't call it in core

Reply:
exactly, that's what I wanted to say

Reply:
rest doesn't have these functions, because so many builder have different instances, we might should make a single builder impl for all commands and then use interfaces, like it is done with Global/guild but that is out of scope for this pr